### PR TITLE
Made `positions` render joined with data for ad and for matching

### PR DESCRIPTION
**Pull request by siefkenj**
The data stored in `position`, `position_data_for_matching` and `position_data_for_ad` should all be joined when rendered. This PR creates a new `PositionService` to help create and get all needed data associated with a position.

**TODO** This PR does not yet handle the `instructor_ids` field correctly.

### CODE REVIEW

**Comment by siefkenj** [comment]
Merging for now so we can continue with test development. Still need a review, though!

File: backend/app/models/assignment.rb [hunk 1]
```diff
@@ -15,6 +15,11 @@ class Assignment < ApplicationRecord
     validates_uniqueness_of :applicant_id, scope: [:position_id]
 
     scope :by_position, ->(position_id) { where(position_id: position_id).order(:id) }
+    scope(:by_session, lambda do |session_id|
```
Comment (tsimica) [inline backend/app/models/assignment.rb:18]: Keep one style or the other (line 17 vs. 18).

File: backend/app/controllers/concerns/transaction_handler.rb [hunk 2]
```diff
@@ -7,7 +7,15 @@ def start_transaction_and_rollback_on_exception
         ActiveRecord::Base.transaction do
             yield
         rescue StandardError => e
-            render_error(message: e.message)
+            begin
+                render_error(message: e.message)
+            rescue NoMethodError
```
Comment (tsimica) [inline backend/app/controllers/concerns/transaction_handler.rb:12]: Is this the only error that can happen? I would leave it as rescue from StandardError and let it bubble up.

File: backend/app/controllers/api/v1/admin/positions_controller.rb [hunk 3]
```diff
@@ -1,31 +1,55 @@
...
 
     # POST /positions/delete
     def delete
-        render_success(object: @position, condition: proc { @position.destroy! })
+        render_on_condition(object: @position, condition: proc { @position.destroy! })
     end
 
     private
 
+    # This method may be manually called from other controllers. Because
```
Comment (tsimica) [inline backend/app/controllers/api/v1/admin/positions_controller.rb:18]: I like the refactoring 😄

File: backend/app/services/position_service.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    end
+
+    def perform
+        @position = Position.new(position_params)
+        @position.save!
+        @position.instructor_ids = instructor_id_params if instructor_id_params
+        @position_data_for_ad = @position.create_position_data_for_ad(position_data_for_ad_params)
+        @position_data_for_matching = @position.create_position_data_for_matching(position_data_for_matching_params)
+        @position_data_for_matching.save!
+        @position_data_for_ad.save!
```
Comment (tsimica) [inline backend/app/services/position_service.rb:21]: Transaction might be necessary here? I see multiple saves going on.

File: backend/app/services/position_service.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    end
+
+    def update(params:)
+        @params = params
+        @position.update!(position_params)
+        @position.instructor_ids = instructor_ids if instructor_ids
+        @position_data_for_ad.update!(position_data_for_ad_params)
+        @position_data_for_matching.update!(position_data_for_matching_params)
```
Comment (tsimica) [inline backend/app/services/position_service.rb:29]: Transaction might be necessary here? Same concern as above.

File: backend/app/controllers/api/v1/admin/session_positions_controller.rb [hunk 6]
```diff
@@ -10,31 +10,17 @@ def index
...
-
-    def position_find_or_create_params
-        params.permit(:id, :position_code, :position_title, :hours_per_assignment,
-                      :start_date, :end_date, :duties, :qualifications,
-                      :ad_hours_per_assignment, :ad_num_assignments,
-                      :ad_open_date, :ad_close_date, :desired_num_assignments,
-                      :current_enrollment, :current_waitlisted, :instructor_ids)
+    def find_position
+        find_session
+        @position = @session.positions.find_by_id(params[:id])
```
Comment (tsimica) [inline backend/app/controllers/api/v1/admin/session_positions_controller.rb:20]: This should be `find_by(id: params[:id])`

File: backend/app/models/assignment.rb (hunk 1 above)
Comment (siefkenj) [reply backend/app/models/assignment.rb:18]: Rubocop doesn't like multiline lambdas...

File: backend/app/services/position_service.rb (hunk 4 above)
Comment (siefkenj) [reply backend/app/services/position_service.rb:21]: The whole action is called from within a transaction when it is used. But...maybe someone will misuse it?

File: backend/app/controllers/api/v1/admin/session_positions_controller.rb (hunk 6 above)
Comment (siefkenj) [reply backend/app/controllers/api/v1/admin/session_positions_controller.rb:20]: Why is that?

Comment (tsimica) [reply backend/app/controllers/api/v1/admin/session_positions_controller.rb:20]: I'm surprised that it works in Rails 6. `find_by_<attribute>` was changed in Rails 5 to `find_by(attribute: x)`.

File: backend/app/services/position_service.rb (hunk 4 above)
Comment (tsimica) [reply backend/app/services/position_service.rb:21]: Makes sense. No need for it then.

Comment (reidka) [reply backend/app/services/position_service.rb:21]: I was expecting @tsimica to take issue with this class.   I don't understand enough about how ruby or rails works, but I was wondering if Position should have position_data_for_ad and position_data_for_matching as nested attributes or delegates.

I find this approach pretty readable though.

**Pull request merged**